nixos/modules/services/web-apps/weblate.nix

```diff
@@ -0,0 +1,398 @@
+  };
+  environment = {
+    PYTHONPATH = "${settings_py}";
+    DJANGO_SETTINGS_MODULE = "settings";
+    GI_TYPELIB_PATH = "${pkgs.pango.out}/lib/girepository-1.0:${pkgs.harfbuzz}/lib/girepository-1.0";
+  };
+  weblate-env = pkgs.writeShellScriptBin
+    "weblate-env"
+    ((pkgs.lib.concatStrings (pkgs.lib.mapAttrsToList (n: v: "export ${n}=${v}\n") environment)) + ''
+      eval -- "\$@"
+    '');
+  weblatePath = with pkgs; [
+        gitSVN
+
+        #optional
+        git-review
+        tesseract
+        licensee
+        mercurial
+      ];
+in
+{
+
+  options = {
+    services.weblate = {
+      enable = lib.mkEnableOption "Weblate service";
+
+      localDomain = lib.mkOption {
+        description = "The domain serving your Weblate instance.";
+        example = "weblate.example.org";
+        type = lib.types.str;
+      };
+
+      djangoSecretKeyFile = lib.mkOption {
+        description = ''
+          Location of the Django secret key.
+
+          This should be a string, not a nix path, since nix paths are copied into the world-readable nix store.
+        '';
+        type = lib.types.path;
+      };
+
+      extraConfig = lib.mkOption {
+        type = lib.types.lines;
+        default = "";
+        description = ''
+          Text to append to <filename>settings.py</filename> Weblate config file.
+        '';
+      };
+
+      smtp = {
+        user = lib.mkOption {
+          description = "SMTP login name.";
+          example = "weblate@weblate.example.org";
+          type = lib.types.str;
+        };
+
+        host = lib.mkOption {
+          description = "SMTP host used when sending emails to users.";
+          type = lib.types.str;
+          default = "127.0.0.1";
+        };
+
+        createLocally = lib.mkOption {
+          description = "Configure local Postfix SMTP server for Weblate.";
+          type = lib.types.bool;
+          default = true;
+        };
+        passwordFile = lib.mkOption {
+          description = ''
+            Location of a file containing the SMTP password.
+
+            This should be a string, not a nix path, since nix paths are copied into the world-readable nix store.
+          '';
+          type = lib.types.path;
+        };
+      };
+
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+
+    assertions = [
+      {
+        assertion = cfg.smtp.createLocally -> cfg.smtp.host == "127.0.0.1";
+        message = ''services.weblate.smtp.host should be "127.0.0.1" if you want to to use services.weblate.smtp.createLocally.'';
+      }
+    ];
+
+    services.nginx = {
+      enable = true;
+      virtualHosts."${cfg.localDomain}" = {
+
+        forceSSL = true;
+        enableACME = true;
+
+        locations = {
+          "= /favicon.ico".alias = "${pkgs.weblate}/lib/${pkgs.python3.libPrefix}/site-packages/weblate/static/favicon.ico";
+          "/static/".alias = "${pkgs.weblate}/lib/${pkgs.python3.libPrefix}/site-packages/weblate/static/";
```

Comment:
I'm not sure I understand your proposition? The `/` route must already be served by the uwsgi unix domain socket, so it's necessary to explicitly route only `/static/`.

Reply:
@erictapen 

I meant it would be nice to have `pkgs.weblate.static` which references to `${pkgs.weblate}/lib/${pkgs.python3.libPrefix}/site-packages/weblate/static/` and have in the module system `cfg.staticPackage` which points to `pkgs.weblate.static` and here you would reference `${cfg.staticPackage}/favicon.ico` for example.

So (1) someone can access to the static part of Weblate (here, it means to build the weblate derivation, so few benefits) (2) someone can replace the static part of Weblate with a patch or something easily.

Reply:
Now I get what you are aiming at, but I don't see the benefits yet.

Is this pattern used anywhere else in Nixpkgs you could point me to?

Is there any existing replacement for the static part of Weblate? I don't see the use case.

Reply:
A rapid grep shows at least:
```
pkgs/servers/misc/navidrome/default.nix
26:  ui = callPackage ./ui {

pkgs/servers/gotify/default.nix
34:  ui = callPackage ./ui.nix { };
```
I imagine those are more involved, as they are most likely Node-generated static files.
Maybe `static` is not a good name as it could get confused with a statically-linked binary from what I see in nixpkgs, but `ui` is often used.

> Is there any existing replacement for the static part of Weblate? I don't see the use case.

There may be situations where you want to do `pkgs.runCommand "patch-static" ''cp -r ${pkgs.weblate.static} $out && do something with $out to patch it, e.g. a robots.txt, etc.` and pass it to `cfg.staticPackage` rather than patching the whole `pkgs.weblate`.

I have no strong feeling on this, it can be added later on if anyone has a real need for it, but most of the time, I see that people would appreciate having override options rather than locking them into a `let` expression ; in this case, overlaying the original weblate package would do the job as long as it is a reasonable patch.